src/Prefetch.cpp

```diff
@@ -220,8 +206,122 @@ class InjectPrefetch : public IRMutator {
         } else {
             stmt = op;
         }
+
+        current_func = old_func;
+        stage = old_stage;
+    }
+};
+
+class ReducePrefetchDimension : public IRMutator {
+    using IRMutator::visit;
+
+    size_t max_dim;
+
+    void visit(const Evaluate *op) {
+        IRMutator::visit(op);
+        op = stmt.as<Evaluate>();
+        internal_assert(op);
+        const Call *call = op->value.as<Call>();
+
+        // Reduce the prefetch dimension if bigger than 'max_dim'.
```

Comment:
Move this comment to right above this class, and also expand it a bit: add that it replaces all of the dimensions above max_dim with for loops.